File: app/src/main/java/com/cellblock70/popularmovies/UI/MovieList/MovieViewModel.java
package com.cellblock70.popularmovies.UI.MovieList;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.cellblock70.popularmovies.data.MovieRepository;
import com.cellblock70.popularmovies.data.database.CompleteMovie;
import com.cellblock70.popularmovies.data.database.Movie;

import java.util.List;

public class MovieViewModel extends AndroidViewModel {

    //private MovieDatabase movieDatabase;
    private LiveData<List<Movie>> allMovies;
    private MutableLiveData<List<CompleteMovie>> favoriteMovies;
    private MovieRepository movieRepository;
    LiveData<List<CompleteMovie>> favorites = favoriteMovies;

    public MovieViewModel(@NonNull Application application) {
        super(application);
        movieRepository = MovieRepository.provideRepository(application);
        favoriteMovies = getFavorites();
    }

    public LiveData<List<Movie>> getMovies(String movieListType) {
        allMovies = movieRepository.getMovies(movieListType);
        return allMovies;
    }

    public MutableLiveData<List<CompleteMovie>> getFavorites() {
        if (favoriteMovies == null) {
            favoriteMovies = movieRepository.getFavorites();
        }
        return favoriteMovies;
    }
}

File: app/src/main/java/com/cellblock70/popularmovies/ui/movielist/MovieViewModel.kt
package com.cellblock70.popularmovies.ui.movielist

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.cellblock70.popularmovies.data.database.Movie
import com.cellblock70.popularmovies.domain.MovieRepository
import com.cellblock70.popularmovies.domain.usecase.GetSelectedTabUseCase
import dagger.assisted.Assisted
import dagger.assisted.AssistedFactory
import dagger.assisted.AssistedInject
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch

@HiltViewModel(assistedFactory = MovieViewModel.MovieViewModelFactory::class)
class MovieViewModel @AssistedInject constructor(
    private val movieRepository: MovieRepository,
    private val getSelectedTabUseCase: GetSelectedTabUseCase,
    @Assisted("language") val language: String
) : ViewModel() {

    @AssistedFactory
    interface MovieViewModelFactory {
        fun create(
            @Assisted("language") language: String
        ): MovieViewModel
    }

    val movies: StateFlow<List<Movie>> = movieRepository.getMovies()

    init {
        getMovies()
    }

    private fun getMovies() {
        viewModelScope.launch(Dispatchers.IO) {
            val movieListType = getSelectedTabUseCase()
            movieRepository.getMovies(movieListType , 1, language)
        }
    }
}